### PR TITLE
Add a cop for detecting legacy powershell methods

### PR DESCRIPTION
**Pull request by tas50**
We shouldn't use these anywhere now that we have the improved methods

Signed-off-by: Tim Smith <tsmith@chef.io>

### CODE REVIEW

File: config/chefstyle.yml [hunk 1]
```diff
@@ -713,4 +713,9 @@ Chef/Ruby/GemspecRequireRubygems:
...
   Description: net/https is deprecated and just includes net/http and openssl. We should include those directly instead
   Enabled: true
-  VersionAdded: '1.3.0'
\ No newline at end of file
+  VersionAdded: '1.3.0'
+
+Chef/Ruby/LegacyPowershellOutMethods:
+  Description: Use powershell_exec!/powershell_exec instead of the slower legacy powershell_out!/powershell_out methods.
+  Enabled: true
+  VersionAdded: '1.6.0'
```
Comment (jaymzh) [inline config/chefstyle.yml:721]: why does your editor not properly end files? :(

**Review by jaymzh — APPROVED** [review]
seems fine. No autocorrect?

**Comment by tas50** [comment]
It's not really something we can autocorrect since it returns a totally different type of data. You have to refactor (for the better) the code to use the new method.

File: config/chefstyle.yml (hunk 1 above)
Comment (tas50) [reply config/chefstyle.yml]: I turned it on now just for you. It causes a lot of issues with other projects that get grumpy about newlines

File: lib/rubocop/cop/chef/ruby/legacy_powershell_out_methods.rb [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+          RESTRICT_ON_SEND = %i{powershell_out! powershell_out}.freeze
+
+          def on_send(node)
+            add_offense(node, message: MSG, severity: :refactor)
+          end
+        end
+      end
+    end
+  end
+end
```
Comment (jaymzh) [inline lib/rubocop/cop/chef/ruby/legacy_powershell_out_methods.rb:36]: ::cough::  ;)

Comment (tas50) [reply lib/rubocop/cop/chef/ruby/legacy_powershell_out_methods.rb]: Fixed

**Pull request merged**